dictcc.el

```diff
@@ -271,11 +302,23 @@ At the moment they are of the form `<tr id='trXXX'></tr>'."
                                (dictcc--insert-candidate #'cdr))))))
     (helm :prompt "Filter: " :sources (list source))))
 
+(defun dictcc--select-sourcelang ()
+  "Select source language with completing read."
+  (let ((lang (completing-read "Source language: "  dictcc-source-languages-alist)))
+    (cdr (assoc lang dictcc-source-languages-alist))))
+
+(defun dictcc--select-destination-lang ()
+  "Select destination language with completing read"
+  (let ((lang (completing-read "Destination language: " dictcc-source-languages-alist)))
+    (cdr (assoc lang dictcc-source-languages-alist))))
+
 ;;;###autoload
-(defun dictcc (query)
+(defun dictcc (query &optional ask-languages)
   "Search dict.cc for QUERY and insert a result at point."
-  (interactive "sQuery: \n")
-  (dictcc--request query))
+  (interactive "sQuery: \np")
+  (let* ((dictcc-source-lang (if ask-languages (dictcc--select-sourcelang) dictcc-source-lang))
```

Comment:
`let` is enough here. `let*` is only needed when later definitions use earlier ones. Also if I have the definition of `(interactive "p")` correctly in mind, `ask-languages` will be `1` if there is no prefix argument, so this line will always ask for the source language. Instead I would check explicitly for the value `4` as you are doing it with `16`.

Reply:
Yes, that was what was happening and I have been able to remedy it exactly the way you suggested.